src/main.rs

```diff
@@ -653,30 +708,30 @@ impl Fixer {
         data: &[u8],
         interner: &mut Interner,
         func_infos: &mut Vec<FuncInfo>,
+        arch: Arch,
     ) -> Result<()> {
         // Although we use `goblin` to iterate through the symbol
         // table, we use `symbolic` to read the debug info from the
         // object/archive, because it's easier to use.
         let archive = Archive::parse(&data)
             .with_context(|| format!("parse `{}` referenced by", file_name))?;
 
-        // Get the x86-64 object from the archive, which might be a fat binary.
-        // (On Mac, Firefox is only available on x86-64.)
-        let mut x86_64_object = None;
+        // Get the object of the wanted arch from the archive, which might be a fat binary.
+        let mut the_object = None;
         for object in archive.objects() {
             let object = object.with_context(|| {
                 format!("parse fat binary entry in `{}` referenced by", file_name)
             })?;
-            if object.arch() == Arch::Amd64 {
-                x86_64_object = Some(object);
+            if object.arch() == arch {
```

Comment:
Not an issue, but this has me idly wondering if we can rely on the subarchitecture to match this strictly between the header and objects. Probably?

(The old code that hardcoded the Arch was technically just as """bad""" for this... although in a slightly different way...)